crates/symbolicator/src/services/symcaches.rs

```diff
@@ -230,8 +222,11 @@ impl CacheItemRequest for FetchSymCacheInternal {
     }
 
     fn should_load(&self, data: &[u8]) -> bool {
+        let had_symbolmap = data.ends_with(SYMBOLMAP_MARKER);
         SymCache::parse(data)
-            .map(|symcache| symcache.is_latest())
+            .map(|symcache| {
+                symcache.is_latest() && had_symbolmap == self.bcsymbolmap_handle.is_some()
```

Comment:
This will also recompute if previously there was a symbolmap but now we didn't fetch one.  this can happen because the derived cache lives longer than the download cache.  Not sure how serious an issue it is.